src/Models/HydrostaticFreeSurfaceModels/split_explicit_free_surface.jl

```diff
@@ -132,55 +167,128 @@ function SplitExplicitAuxiliary(grid::AbstractGrid)
+=================
+
+- `substeps`: The number of substeps that divide the range `(t, t + 2Δt)`. NOTE: not all averaging functions
+              require to substep till `2Δt`. The number of substeps will be reduced automatically to the last index
+              of `averaging_weights` for which `averaging_weights > 0`
+- `averaging_function`: function of `τ` used to average `U` and `η` within the barotropic advancement.
+                        `τ` is the fractional substep going from 0 to 2 with the baroclinic time step `t + Δt`
+                        located at `τ = 1`. This function should be centered at `τ = 1` (i.e. ∑(aₘ⋅m/M) = 1)
+- `timestepper`: Time stepping scheme used, either `ForwardBackwardScheme` or `AdamsBashforth3Scheme`
+"""
+function SplitExplicitSettings(; substeps = 200, 
+                                 averaging_function = averaging_shape_function,
+                                 timestepper = ForwardBackwardScheme())
+
+    τᶠ = range(0.0, 2.0, length = substeps+1)
+    Δτ = τᶠ[2] - τᶠ[1]
+
+    averaging_weights = averaging_function.(τᶠ[2:end]) 
+    idx = searchsortedlast(averaging_weights, 0.0, rev=true)
+    substeps = idx
+
+    averaging_weights = averaging_weights[1:idx]
+    mass_flux_weights = similar(averaging_weights)
+    
+    M = searchsortedfirst(τᶠ, 1.0) - 1
+
+    averaging_weights ./= sum(averaging_weights)
+
+    for i in substeps:-1:1
+        mass_flux_weights[i] = 1 / M * sum(averaging_weights[i:substeps]) 
+    end
+
+    mass_flux_weights ./= sum(mass_flux_weights)
+
+    return SplitExplicitSettings(substeps,
+                                 averaging_weights,
+                                 mass_flux_weights, 
+                                 Δτ, 
+                                 timestepper)
 end
 
 # Convenience Functions for grabbing free surface
 free_surface(free_surface::SplitExplicitFreeSurface) = free_surface.η
 
 # extend 
-@inline explicit_barotropic_pressure_x_gradient(i, j, k, grid, ::SplitExplicitFreeSurface) = 0
-@inline explicit_barotropic_pressure_y_gradient(i, j, k, grid, ::SplitExplicitFreeSurface) = 0
+@inline explicit_barotropic_pressure_x_gradient(i, j, k, grid, ::SplitExplicitFreeSurface) = zero(grid)
+@inline explicit_barotropic_pressure_y_gradient(i, j, k, grid, ::SplitExplicitFreeSurface) = zero(grid)
 
 # convenience functor
 function (sefs::SplitExplicitFreeSurface)(settings::SplitExplicitSettings)
     return SplitExplicitFreeSurface(sefs.η, sefs.state, sefs.auxiliary, sefs.gravitational_acceleration, settings)
 end
 
+Base.summary(sefs::SplitExplicitFreeSurface) = string("SplitExplicitFreeSurface with $(sefs.settings.substeps) steps")
+
+Base.show(io::IO, sefs::SplitExplicitFreeSurface) = print(io, "$(summary(sefs))\n")
+
+function reset!(sefs::SplitExplicitFreeSurface)
+    for name in propertynames(sefs.state)
+        var = getproperty(sefs.state, name)
+        fill!(var, 0.0)
+    end
+    fill!(sefs.auxiliary.Gᵁ, 0.0)
+    fill!(sefs.auxiliary.Gⱽ, 0.0)
```

Comment:
```suggestion
    fill!(sefs.auxiliary.Gᵁ, 0)
    fill!(sefs.auxiliary.Gⱽ, 0)
```

this will promote `0` to the type of `sefs.auxiliary.Gᵁ`. Since type promotion has almost no cost, this is efficient.

Using lower types like integer, which will be promoted to the given floating point type, is ok. But we want to avoid the opposite --- using _higher_ floating point types than the eltype of the `grid`.